pkg/controller/operators/olm/operator.go

```diff
@@ -349,6 +398,22 @@ func (a *Operator) namespaceAddedOrRemoved(obj interface{}) {
 			}
 		}
 	}
+	// find any API services owned by this namespace and requeue them
```

Comment:
When a namespace gets deleted with a CSV in it, I'm pretty sure we get `delete` events for CSVs. I think it would be simpler to just queue up apiservices on CSV delete.

Reply:
This is covering the case of deleting the namespace that also contains OLM. There already exists code to handle cleaning up api services upon CSV deletion in the "normal" case:
https://github.com/operator-framework/operator-lifecycle-manager/blob/5159b0a1c0dfe2cb76eb706afb4e3cc2ac4447fd/pkg/controller/operators/olm/operator.go#L449
I assume what happens is upon deleting the namespace hosting OLM, OLM doesn't have time to run any clean up.